tools/xdp/ebpf/src/main.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_std]
+#![no_main]
+
+use aya_bpf::{
+    bindings::xdp_action,
+    macros::{map, xdp},
+    maps::HashMap,
+    programs::XdpContext,
+};
+use s2n_quic_core::xdp::{
+    bpf::DecoderBufferMut,
+    decoder::{decode_packet_validator, Validator},
+};
+
+mod xskmap;
+
+#[map(name = "S2N_QUIC_XDP_SOCKETS")]
+static SOCKETS: xskmap::XskMap = xskmap::XskMap::with_max_entries(1024, 0);
+
+#[map(name = "S2N_QUIC_XDP_PORTS")]
+static PORTS: HashMap<u16, u8> = HashMap::with_max_entries(1024, 0);
+
+#[xdp(name = "s2n_quic_xdp")]
+pub fn s2n_quic_xdp(ctx: XdpContext) -> u32 {
+    let action = handle_packet(&ctx);
+
+    #[cfg(feature = "trace")]
+    {
+        use aya_log_ebpf as log;
+        match action {
+            xdp_action::XDP_DROP => log::trace!(&ctx, "ACTION: DROP"),
```

Comment:
Hmm, I guess this log is stored in a map? Or emitted to a file via `bpf_trace_printk`

Reply:
It uses a pair of maps: https://github.com/aya-rs/aya/blob/main/bpf/aya-log-ebpf/src/lib.rs.

But note that this isn't enabled by default and is really only for debugging.